File: src/1302-Deepest-Leaves-Sum/1302.go
func deepestLeavesSum(root *TreeNode) int {
    q := []*TreeNode{root}
    res := 0
    
    for len(q) > 0 {
        n := len(q)
        res = 0
        for i := 0; i < n; i++ {
            cur := q[0]
            q = q[1:]
            res += cur.Val
            if cur.Left != nil {
                q = append(q, cur.Left)
            }
            if cur.Right != nil {
                q = append(q, cur.Right)
            }
        }
    }
    return res
}

File: src/1302-Deepest-Leaves-Sum/1302.java
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        Queue<TreeNode> q = new LinkedList();
        q.offer(root);
        int res = 0;
        
        while (!q.isEmpty()) {
            int n = q.size();
            res = 0;
            
            while (n-- > 0) {
                TreeNode cur = q.poll();
                res += cur.val;
                if (cur.right != null) q.offer(cur.right);
                if (cur.left != null) q.offer(cur.left); 
            }
        }
        return res;
    }
}